torch_xla/experimental/custom_kernel_from_jax.py

```diff
@@ -0,0 +1,397 @@
+import dataclasses
+import functools
+import json
+from dataclasses import asdict
+from typing import Any
+
+import torch
+import torch_xla.debug.profiler as xp
+from torch.library import custom_op
+from torch.utils._pytree import tree_flatten
+from torch_xla.core.xla_builder import call_jax
+from torch_xla.distributed.spmd import Mesh
+from torch_xla.experimental.custom_kernel import requires_jax
+
+
+@dataclasses.dataclass
```

Comment:
We can add eq=True and hash=True to the dataclasses decorator call. This way any instance of this config will be hashable and can be used as an argument to call_jax, avoiding the lru cache.